asf_heat_pump_suitability/pipeline/transform/non_residential_entities.py

```diff
@@ -66,12 +81,35 @@ def transform_gdf_non_residential_buildings(
     exclude_buildings_gdf = important_building_gdf[
         important_building_gdf[col].isin(NO_RESIDENTIAL_OVERLAP_BUILDING_TYPES)
     ]
+    poi_buildings_gdf = building_gdf.sjoin(poi_gdf, how="inner", predicate="contains")
 
     # Get buildings which are railway stations (railway stations are only given as point geometries)
     railway_station_gdf = building_gdf.sjoin(
         railway_station_gdf, how="inner", predicate="contains"
     )
 
-    return pd.concat(
-        [exclude_buildings_gdf[["geometry"]], railway_station_gdf[["geometry"]]]
+    exclude_buildings_gdf = pd.concat(
+        [
+            exclude_buildings_gdf[["geometry"]],
+            railway_station_gdf[["geometry"]],
+            poi_buildings_gdf[["geometry"]],
+        ]
+    )
+
+    # Get locations of UPRNs in domestic EPC
+    include_uprns = uprns.load_set_valid_epc_uprns(epc_type="domestic")
+    uprns_gdf = uprns_gdf[uprns_gdf["UPRN"].isin(include_uprns)].copy()
+
+    # Find buildings that contain a domestic UPRN
+    exclude_buildings_gdf = exclude_buildings_gdf.sjoin(
+        uprns_gdf[["UPRN", "geometry"]], how="left", predicate="contains"
     )
+
+    # Filter to buildings that don't contain a domestic UPRN
+    exclude_buildings_gdf = exclude_buildings_gdf[
+        exclude_buildings_gdf["UPRN"].isnull()
+    ].drop(columns=["UPRN", "index_right"])
+
+    # Normalize to drop duplicate building footprints
+    exclude_buildings_gdf["geometry"] = exclude_buildings_gdf.normalize()
+    return exclude_buildings_gdf.drop_duplicates()
```

Comment:
why are geometries different for the same building?
and wouldn't it be enough to drop by building ID, if there is one in this df?